protocol/strict-derivation.md

```diff
@@ -0,0 +1,269 @@
+out later by a backwards-invalidating singular batch derived from a span batch. So if we only
+forwards-invalidate, this becomes easier.
+
+And in any case, a restart or reorg needs to reconstruct the derivation pipeline state from the new
+(sync-start) L1 block, so I don't understand how reconstructing span batch derivation state is
+different from reconstructing the channel bank and batch queue in general. If a restart or reorg
+happens in the middle of deriving singular batches from a channel, we're in a similar situation that
+is solved by going back far enough for the sync-start.
+
+## Proposed Design
+
+To resolve open questions regarding the implementation of Partial Span Batch Validity, I propose to
+apply the principle to consistently change the validation rules to be only forward-invalidating, and
+thus treat a span batch like an additional (optional) stage in the derivation pipeline that sits
+before the batch queue, so that the batch queue pulls singular batches from this previous span batch
+stage. This way, we never backwards-invalidate the span batch, only forward.
+
+This means that sequencer drift validations are applied individually to each singular batch derived
+from the span batch. On a failed check, due to Steady Batch Derivation, the singular batch is
+immediately derived as a deposit-only block, and the remaining span batch is invalidated.
+
+There are still checks that naturally apply to the whole span batch due to its data structure:
+- A span batch only has one parent reference, since all internal batch references are implicit.
+Therefore, a failed parent check invalidates the whole span batch.
+- A failed sequencing window check will always apply to the oldest batches in a span batch, so a
+failed check of the first batch will invalidate the full span batch.
+- If `span_end.timestamp < next_timestamp` (as defined in the Span Batch specs), the whole batch is
+dropped, as it doesn't contain any new batches (this would also happen if applying timestamp checks
+to each derived singular batch individually).
+
+Those checks are done once per span batch before the batch queue derives any singular batches.
+
+# Strict Batch Ordering
+
+Strict batch ordering requires batches within and across channels to be strictly ordered.
+This removes the necessity to buffer more than one channel, and the channel bank effectively holds
+at most one staging channel, for which it collects frames, that must arrive in order.
+
+The following will explore the open design space, presenting options, and propose a solution.
+The principle I would apply to answer open design questions is that the new rules should generally
+choose the design that leads to faster progess of derivation, even if it means deriving
+empty batches earlier.
+Spoiler: Option 1 will always be the proposed option.
+
+## Out of order frames
+
+There's an open design space around how to handle some scenarios for missing or out of order frames:
```

Comment:
Yes, batcher restarts can in theory cause the batcher to not close an open channel and then submit a new one.
Then there's also the scenario where a batcher cannot get a tx on-chain and requeues the blocks and then attempts sending them again in a new channel. Depending on the specifics of the batcher implementation, this can also lead to out of order frames. All quite unlikely and almost never happening, but we need to take extra care to harden the batcher against such behavior in the future.